tst/EnergyPlus/unit/FuelCellElectricGenerator.unit.cc

```diff
@@ -55,132 +55,1027 @@
+    // EXPECT_EQ("PLANT LOOP 1", state->dataPlnt->PlantLoop(generatorController->cogenLocation.loopNum).Name);
+
+    EXPECT_EQ("GENERATOR FUEL CELL EXHAUST GAS TO WATER HEAT EXCHANGER 1", generatorController->compPlantName);
+    EXPECT_EQ(0, generatorController->generatorIndex);
+
+    EXPECT_EQ("ALWAYS ON DISCRETE", generatorController->availSched);
+    EXPECT_GT(generatorController->availSchedPtr, 0);
+
+    EXPECT_EQ(0, generatorController->nominalThermElectRatio);
+
+    EXPECT_TRUE(generatorController->onThisTimestep);
+
+    // FC has 3400 of power per input, and operates in Baseload mode
+    EXPECT_EQ(3400.0, generatorController->maxPowerOut);
+    EXPECT_EQ(3400.0, generatorController->powerRequestThisTimestep);
+    EXPECT_EQ(3400.0, generatorController->electProdRate);
+    EXPECT_EQ(3400.0*15*60, generatorController->electricityProd); // Timestep = 4, so 15min
+    EXPECT_EQ(0, generatorController->dCElectricityProd);
+    EXPECT_EQ(0, generatorController->dCElectProdRate);
+
+
+
+    auto thisFCcompPtr = FuelCellElectricGenerator::FCDataStruct::factory(*state, generatorController->name);
+    auto thisFC = dynamic_cast<FuelCellElectricGenerator::FCDataStruct*> (thisFCcompPtr);
+    // Power Module
+    EXPECT_EQ("GENERATOR FUEL CELL POWER MODULE 1", thisFC->NameFCPM);
+    auto fCPM = thisFC->FCPM;
+    EXPECT_EQ(fCPM.Name, thisFC->NameFCPM);
+    // Annex42 = Direct
+    EXPECT_EQ(DataGenerators::CurveMode::Direct, fCPM.EffMode);
+    ASSERT_GT(fCPM.EffCurveID, 0);
+    EXPECT_EQ("POWER MODULE EFFICIENCY CURVE", state->dataCurveManager->PerfCurve(fCPM.EffCurveID).Name);
+    EXPECT_EQ(1.0, fCPM.NomEff);
+    EXPECT_EQ(3400.0, fCPM.NomPel);
+    // At beggining of simulation, 10. But then it was simulated, so it should be greater
+    EXPECT_GT(fCPM.NumCycles, 10);
+    EXPECT_EQ(0.0, fCPM.CyclingDegradRat);
+    EXPECT_EQ(1.0, fCPM.NomEff);
+    EXPECT_EQ(3400.0, fCPM.NomPel);
+    EXPECT_EQ(0.0, fCPM.NumCycles);
+    EXPECT_EQ(0.0, fCPM.CyclingDegradRat);
+    EXPECT_EQ(0.0, fCPM.NumRunHours);
+    EXPECT_EQ(0.0, fCPM.OperateDegradRat);
+    EXPECT_EQ(10000.0, fCPM.ThreshRunHours);
+    EXPECT_EQ(1.4, fCPM.UpTranLimit);
+    EXPECT_EQ(0.2, fCPM.DownTranLimit);
+    EXPECT_EQ(0.0, fCPM.StartUpTime);
+    EXPECT_EQ(0.2, fCPM.StartUpFuel);
+    EXPECT_EQ(0.0, fCPM.StartUpElectConsum);
+    EXPECT_EQ(0.0, fCPM.StartUpElectProd);
+    EXPECT_EQ(0.0, fCPM.ShutDownTime);
+    EXPECT_EQ(0.2, fCPM.ShutDownFuel);
+    EXPECT_EQ(0.0, fCPM.ShutDownElectConsum);
+    EXPECT_EQ(0.0, fCPM.ANC0);
+    EXPECT_EQ(0.0, fCPM.ANC1);
+
+    EXPECT_EQ(DataGenerators::SkinLoss::ConstantRate, fCPM.SkinLossMode);
+
+    EXPECT_EQ("THERMAL ZONE 1", fCPM.ZoneName);
+    EXPECT_EQ(1, fCPM.ZoneID);
+
+    EXPECT_EQ(0.6392, fCPM.RadiativeFract);
+
+    // TODO: this is an actual bug. It is being overriden in FigureFuelCellZoneGains, while here the SkinLossMode is ConstantRate, so it should be
+    // unaffected throughout
+    EXPECT_EQ(729, fCPM.QdotSkin);
```

Comment:
TODO: this seems like a bug.

It's being overriden to 0.0 here: https://github.com/NREL/EnergyPlus/blob/f754bc5c2c5d415789c9b01297c575772833b862/src/EnergyPlus/FuelCellElectricGenerator.cc#L3343

Reply:
Addressed it in https://github.com/NREL/EnergyPlus/pull/8511/commits/4f837e59b1648cb73aeeb7af39dfb81ea96affcd